FireboltDotNetSdk.Tests/Integration/DataTypesTest.cs

```diff
@@ -331,6 +334,199 @@ public void ExecuteStreamedQueryWithDifferentDataTypes()
+            // Create table with explicit struct schema
+            var createStruct = (FireboltCommand)connection.CreateCommand();
+            createStruct.CommandText = $"CREATE TABLE struct_various (s struct({structTypeDef}))";
+            createStruct.ExecuteNonQuery();
+
+            // Insert rows by packing helper columns into the struct
+            var structExpr = string.Join(", ", aliases);
+            var insertStruct = (FireboltCommand)connection.CreateCommand();
+            insertStruct.CommandText = $"INSERT INTO struct_various SELECT struct({structExpr}) FROM struct_various_types_helper";
+            insertStruct.ExecuteNonQuery();
+
+            // Validate: read the struct and compare every field to ExpectedValues
+            var select = (FireboltCommand)connection.CreateCommand();
+            select.CommandText = "SELECT s FROM struct_various";
+            using var reader = select.ExecuteReader();
+            Assert.That(reader.Read(), Is.True);
+
+            var standardDic = (Dictionary<string, object?>)reader.GetValue(0);
+            var genericDic = reader.GetFieldValue<Dictionary<string, object?>>(0);
+            Assert.Multiple(() =>
+            {
+                Assert.That(standardDic, Is.Not.Null);
+                Assert.That(standardDic.Keys, Has.Count.EqualTo(aliases.Count));
+                Assert.That(genericDic, Is.Not.Null);
+                Assert.That(genericDic.Keys, Has.Count.EqualTo(aliases.Count));
+            });
+
+            for (var i = 0; i < aliases.Count; i++)
+            {
+                var key = aliases[i];
+                var expected = ExpectedValues[i];
+                Assert.Multiple(() =>
+                {
+                    Assert.That(standardDic.ContainsKey(key), Is.True, $"Missing key '{key}' in struct");
+                    Assert.That(genericDic.ContainsKey(key), Is.True, $"Missing key '{key}' in struct");
+                });
+                var standardActual = standardDic[key];
+                var genericActual = genericDic[key];
+                if (expected is null)
+                {
+                    Assert.Multiple(() =>
+                    {
+                        Assert.That(standardActual, Is.Null, $"Field '{key}' expected null");
+                        Assert.That(genericActual, Is.Null, $"Field '{key}' expected null");
+                    });
+                }
+                else
+                {
+                    Assert.Multiple(() =>
+                    {
+                        Assert.That(standardActual, Is.EqualTo(expected), $"Mismatch at '{key}'");
+                        Assert.That(genericActual, Is.EqualTo(expected), $"Mismatch at '{key}'");
+                    });
+                }
+            }
+
+            // Also deserialize into POCO using FireboltStructNameAttribute and validate
+            var poco = reader.GetFieldValue<VariousStructPoco>(0);
+            var propByAlias = typeof(VariousStructPoco)
```

Comment:
It would be cleaner to make this a separate test. We can always avoid setup duplication by creating a method to do this.